src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/ControlPlaneDataStore.cs

```diff
@@ -90,10 +101,30 @@ public async Task<IdentityProvider> UpsertIdentityProviderAsync(IdentityProvider
         private async Task<T> GetSystemDocumentByIdAsync<T>(string id, string partitionKey, CancellationToken cancellationToken)
         {
             EnsureArg.IsNotNull(id, nameof(id));
+            var response = await GetSystemDocumentsByIdAsync<T>(id, partitionKey, cancellationToken);
+            var result = response.SingleOrDefault();
+            if (result == null)
+            {
+                _logger.LogError($"{typeof(T)} with id {id} was not found in Cosmos DB.");
+            }
+
+            return result;
+        }
+
+        private async Task<IEnumerable<T>> GetSystemDocumentsByIdAsync<T>(string id, string partitionKey, CancellationToken cancellationToken)
+        {
+            FeedResponse<Document> response = await GetSystemDocuments<T>(id, partitionKey, cancellationToken);
+            return response.Select(r => r.GetPropertyValue<T>("r"));
+        }
+
+        private async Task<FeedResponse<Document>> GetSystemDocuments<T>(string id, string partitionKey, CancellationToken cancellationToken)
```

Comment:
Let's go one step more in making this generic and allow it to accept a set of key value pairs that can ben appended to the query. That way we can use this also for a basic search/filter that will be needed for the role memberships.